### ISSUE TITLE
Cluster API endpoints errors when 1 node is remaining in cluster 

### ISSUE THREAD

**Issue by boudey** [hashicorp/vault#22063]
I have asked a question in the discussion forums[ here ](https://discuss.hashicorp.com/t/2-node-vault-cluster-apis-unresponsive-after-killing-follower-node/56504) but further reading into the code makes me think this is a bug

**Describe the bug**
In a cluster, when other peers are shut down and one peer is remaining, several cluster API endpoints are not functioning including `/v1/sys/storage/raft/configuration` and `/v1/sys/ha-status` giving the error: `local node not active but active cluster node not found`

**To Reproduce**
1. Setup a 2-node cluster (I have verified the error with both `retry_join` and `vault operator raft join`
2. start node_0, init and unseal it, verify that `vault operator raft list-peers` returns one node
3. start node_1, unseal it, verify that `vault operator raft list-peers` returns both nodes
```
root@e21fea7c1f99:# vault operator raft list-peers
Node                                    Address             State       Voter
----                                    -------             -----       -----
cf0dc62e-9193-bb64-19a4-5643b6f19517    172.16.0.38:8201    leader      true
8219a3b2-f140-46f4-992d-5a4cf0acf791    172.16.0.43:8201    follower    true
```
4. kill node_1
5. `vault operator raft list-peers` no longer works on node_0 with above error. No new nodes can join the cluster. Basically the cluster is not useful unless node_1 rejoins.

```
root@e21fea7c1f99:# vault operator members
Error making API request.

URL: GET http://172.16.0.38:8200/v1/sys/ha-status
Code: 500. Errors:

* local node not active but active cluster node not found
root@e21fea7c1f99:# vault operator raft list-peers
Error reading the raft cluster configuration: Error making API request.

URL: GET http://172.16.0.38:8200/v1/sys/storage/raft/configuration
Code: 500. Errors:

* local node not active but active cluster node not found

```

**Expected behavior**
`vault operator raft list-peers` does not return error 
other nodes should be able to join the cluster

**Environment:**
* Vault Server Version (retrieve with `vault status`): 1.14.1
* Vault CLI Version (retrieve with `vault version`): Vault v1.14.1 (bf23fe8636b04d554c0fa35a756c75c2f59026c0)
* Server Operating System/Architecture:  Linux 4.15.0-169-generic #177-Ubuntu

Vault server configuration file(s):

on node_0 (172.16.0.38)
```hcl

	storage "raft" {
		path    = "/vault"
	retry_join {
		leader_api_addr = "http://172.16.0.43:8200"
		}

	  }

	  listener "tcp" {
		address     = "127.0.0.1:8200"
		tls_disable = "true"
	  }

	  listener "tcp" {
		address = "172.16.0.38:8200"
		tls_disable = "true"
	  }

	  api_addr = "http://172.16.0.38:8200"
	  cluster_addr = "https://172.16.0.38:8201"
	  disable_mlock = true
	  ui = true
	  log_level = "TRACE"
```
on node_1 (172.16.0.43)
```hcl
	storage "raft" {
		path    = "/vault"
	retry_join {
        leader_api_addr = "http://172.16.0.38:8200"
    }

	  }

	  listener "tcp" {
		address     = "127.0.0.1:8200"
		tls_disable = "true"
	  }

	  listener "tcp" {
		address = "172.16.0.43:8200"
		tls_disable = "true"
	  }

	  api_addr = "http://172.16.0.43:8200"
	  cluster_addr = "https://172.16.0.43:8201"
	  disable_mlock = true
	  ui = true
	  log_level = "TRACE"
```

**Additional context**
I can see that the leader node_0 steps down when node_1 was killed, and keeps trying to vote itself back as leader but fails
```
2023-07-25T22:56:02.045Z [ERROR] storage.raft: failed to heartbeat to: peer=172.16.0.43:8201 backoff time=10ms error="dial tcp 172.16.0.43:8201: connect: connection refused"
2023-07-25T22:56:02.285Z [DEBUG] core.cluster-listener: creating rpc dialer: address=172.16.0.43:8201 alpn=raft_storage_v1 host=raft-82a6ee8c-995f-f593-02a9-7834fdd9478c
2023-07-25T22:56:02.286Z [ERROR] storage.raft: failed to appendEntries to: peer="{Voter 8219a3b2-f140-46f4-992d-5a4cf0acf791 172.16.0.43:8201}" error="dial tcp 172.16.0.43:8201: connect: co
nnection refused"
2023-07-25T22:56:02.627Z [DEBUG] core.cluster-listener: creating rpc dialer: address=172.16.0.43:8201 alpn=raft_storage_v1 host=raft-82a6ee8c-995f-f593-02a9-7834fdd9478c
2023-07-25T22:56:02.627Z [ERROR] storage.raft: failed to heartbeat to: peer=172.16.0.43:8201 backoff time=20ms error="dial tcp 172.16.0.43:8201: connect: connection refused"
2023-07-25T22:56:02.864Z [WARN]  storage.raft: failed to contact: server-id=8219a3b2-f140-46f4-992d-5a4cf0acf791 time=2.500850963s
2023-07-25T22:56:02.864Z [WARN]  storage.raft: failed to contact quorum of nodes, stepping down
2023-07-25T22:56:02.864Z [INFO]  storage.raft: entering follower state: follower="Node at 172.16.0.38:8201 [Follower]" leader-address= leader-id=
2023-07-25T22:56:02.865Z [WARN]  core: leadership lost, stopping active operation
2023-07-25T22:56:02.865Z [INFO]  core: pre-seal teardown starting
2023-07-25T22:56:02.865Z [DEBUG] storage.raft.autopilot: state update routine is now stopped
2023-07-25T22:56:02.865Z [DEBUG] storage.raft.autopilot: autopilot is now stopped
2023-07-25T22:56:03.365Z [INFO]  core: stopping raft active node
2023-07-25T22:56:03.365Z [DEBUG] expiration: stop triggered
2023-07-25T22:56:03.365Z [TRACE] expiration.job-manager: terminating job manager...
2023-07-25T22:56:03.365Z [TRACE] expiration.job-manager: terminating dispatcher
2023-07-25T22:56:03.365Z [DEBUG] expiration: finished stopping
2023-07-25T22:56:03.366Z [INFO]  rollback: stopping rollback manager
2023-07-25T22:56:03.366Z [INFO]  core: pre-seal teardown complete
2023-07-25T22:56:03.366Z [ERROR] core: clearing leader advertisement failed: error="node is not the leader"
2023-07-25T22:56:03.366Z [ERROR] core: unlocking HA lock failed: error="node is not the leader"
2023-07-25T22:56:03.366Z [TRACE] core: found new active node information, refreshing
2023-07-25T22:56:03.404Z [DEBUG] core.cluster-listener: creating rpc dialer: address=172.16.0.43:8201 alpn=raft_storage_v1 host=raft-82a6ee8c-995f-f593-02a9-7834fdd9478c
2023-07-25T22:56:03.405Z [ERROR] storage.raft: failed to heartbeat to: peer=172.16.0.43:8201 backoff time=40ms error="dial tcp 172.16.0.43:8201: connect: connection refused"
2023-07-25T22:56:03.647Z [DEBUG] core.cluster-listener: creating rpc dialer: address=172.16.0.43:8201 alpn=raft_storage_v1 host=raft-82a6ee8c-995f-f593-02a9-7834fdd9478c
2023-07-25T22:56:03.648Z [ERROR] storage.raft: failed to appendEntries to: peer="{Voter 8219a3b2-f140-46f4-992d-5a4cf0acf791 172.16.0.43:8201}" error="dial tcp 172.16.0.43:8201: connect: connection refused"

2023-07-25T22:59:20.573Z [TRACE] core: found new active node information, refreshing
2023-07-25T22:59:23.073Z [TRACE] core: found new active node information, refreshing
2023-07-25T22:59:25.574Z [TRACE] core: found new active node information, refreshing
2023-07-25T22:59:26.788Z [WARN]  storage.raft: Election timeout reached, restarting election
2023-07-25T22:59:26.788Z [INFO]  storage.raft: entering candidate state: node="Node at 172.16.0.38:8201 [Candidate]" term=325
2023-07-25T22:59:26.791Z [DEBUG] storage.raft: voting for self: term=325 id=cf0dc62e-9193-bb64-19a4-5643b6f19517
2023-07-25T22:59:26.795Z [DEBUG] storage.raft: asking for vote: term=325 from=8219a3b2-f140-46f4-992d-5a4cf0acf791 address=172.16.0.43:8201
2023-07-25T22:59:26.795Z [DEBUG] storage.raft: calculated votes needed: needed=2 term=325
2023-07-25T22:59:26.795Z [DEBUG] storage.raft: vote granted: from=cf0dc62e-9193-bb64-19a4-5643b6f19517 term=325 tally=1
2023-07-25T22:59:26.795Z [DEBUG] core.cluster-listener: creating rpc dialer: address=172.16.0.43:8201 alpn=raft_storage_v1 host=raft-82a6ee8c-995f-f593-02a9-7834fdd9478c
2023-07-25T22:59:26.795Z [ERROR] storage.raft: failed to make requestVote RPC: target="{Voter 8219a3b2-f140-46f4-992d-5a4cf0acf791 172.16.0.43:8201}" error="dial tcp 172.16.0.43:8201: connect: connection refused" term=325
```

I am unable to remove the killed peer in this situation
```
root@e21fea7c1f99:~# vault operator raft remove-peer 8219a3b2-f140-46f4-992d-5a4cf0acf791
Error removing the peer from raft cluster: Error making API request.

URL: PUT http://172.16.0.38:8200/v1/sys/storage/raft/remove-peer
Code: 500. Errors:

* local node not active but active cluster node not found
```

New nodes are unable to join the cluster in this state because of the same error
```
root@e2ceae16c340:~# vault operator raft join http://172.16.0.38:8200
Error joining the node to the Raft cluster: Error making API request.

URL: POST http://127.0.0.1:8200/v1/sys/storage/raft/join
Code: 500. Errors:

* failed to join raft cluster: failed to get raft challenge


2023-07-25T23:22:29.610Z [INFO]  core: attempting to join possible raft leader node: leader_addr=http://172.16.0.38:8200
2023-07-25T23:22:29.617Z [ERROR] core: failed to get raft challenge: leader_addr=http://172.16.0.38:8200
  error=
  | error during raft bootstrap init call: Error making API request.
  |
  | URL: PUT http://172.16.0.38:8200/v1/sys/storage/raft/bootstrap/challenge
  | Code: 500. Errors:
  |
  | * local node not active but active cluster node not found

2023-07-25T23:22:29.617Z [ERROR] core: failed to join raft cluster: error="failed to get raft challenge"
```

So basically the cluster is hung unless node_1 rejoins.

In the code I can see in `http/handler.go` that this error is returned if no leader is found, but since node_0 stepped down this error will be returned perpetually with no chance even to remove node_1?

**Comment by maxb** [comment]
I replied in the discuss.hashicorp.com thread.

**Comment by boudey** [comment]
Thank you
